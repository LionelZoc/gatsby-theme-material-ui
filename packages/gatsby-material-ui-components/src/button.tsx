import MuiButton, { ButtonProps } from "@material-ui/core/Button";

import patchButtonBaseComponent from "./patch-base-button-components";

export const Button = patchButtonBaseComponent<ButtonProps>(MuiButton);
