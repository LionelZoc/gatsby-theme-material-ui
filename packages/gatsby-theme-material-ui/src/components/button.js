import React from 'react';
import MuiButton from '@material-ui/core/Button';

import GatsbyLink from './gatsby-link';

function Button(props) {
  const { to, href } = props;
  const component = to || href ? GatsbyLink : 'button';

  return <MuiButton component={component} {...props} />;
}

export default Button;
