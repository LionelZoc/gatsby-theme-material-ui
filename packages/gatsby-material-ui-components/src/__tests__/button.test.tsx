import * as React from 'react';
import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { Button } from '../button';

describe('button', () => {
  beforeEach(() => {
    // gatsby link unit tests crashed when globals are undefined
    (global as any).___loader = {
      enqueue: jest.fn(),
    };
  });

  it('should be rendered correctly', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('lbutton with to', () => {
    const { asFragment } = render(<Button to="/page2" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('button with ref', () => {
    const { result } = renderHook(() => React.useRef<HTMLButtonElement>(null));
    const buttonRef = result.current;
    const { asFragment } = render(<Button ref={buttonRef} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
