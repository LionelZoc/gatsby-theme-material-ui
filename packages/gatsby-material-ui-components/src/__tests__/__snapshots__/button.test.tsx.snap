// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button button with ref 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`button lbutton with to 1`] = `
<DocumentFragment>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    href="/page2"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </a>
</DocumentFragment>
`;

exports[`button should be rendered correctly 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
